metrics/segmentStats.py

```diff
@@ -0,0 +1,506 @@
+
+    def _cyclicTree(self, subGraphSkeleton, cycleList):
+        """
+        Find statistics of a cyclic tree of a disjoint graph
+
+        Parameters
+        ----------
+        subGraphSkeleton: Networkx Graph
+            disjoint subgraph of the skeleton's networkx graph
+
+        cyclesList: list of lists
+            list of list of nodes in the cycles
+
+        cycles: int
+            number of cycles in the graph so far
+        Returns
+        -------
+        subGraphSkeleton : Networkx graph
+            graph changed inplace with visited nodes removed
+
+        cycleInfoDict : Dictionary
+            class instance variable changed in place with key as nth cycle, value as list with
+            [number of branch points on cycle, length of the cycle]
+
+        countDict : Dictionary
+            class instance variable changed in place with key as source node,
+            value is the number of segments attached to the source node
+
+        lengthDict : Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the length of the segment
+
+        tortuosityDict : Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the tortuosity of the segment
+
+        contractionDictdict : Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the contraction of the segment
+
+        hausdorffDimensionDict : Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the hausdorff dimension of the segment
+        """
+        for nthcycle, cycle in enumerate(cycleList):
+            nodeDegreeDictFilt = {key: value for key, value in self._nodeDegreedict.items() if key in cycle}
+            branchPointsOnCycle = [k for (k, v) in nodeDegreeDictFilt.items() if v != 2 and v != 1]
+            sourceOnCycle = branchPointsOnCycle[0]
+            if len(branchPointsOnCycle) == 1:
+                self._singleCycle(subGraphSkeleton, cycle)
+            else:
+                for point in cycle:
+                    if point in branchPointsOnCycle:
+                        if nx.has_path(subGraphSkeleton, source=sourceOnCycle, target=point) and sourceOnCycle != point:
+                            simplePath = nx.shortest_path(subGraphSkeleton, source=sourceOnCycle, target=point)
+                            sortedSegment = sorted(simplePath)
+                            if (sortedSegment not in self._sortedSegments and
+                               sum([1 for pathPoint in simplePath if pathPoint in branchPointsOnCycle]) == 2 and
```

Comment:
What does this part of the if do? The `sum` piece of it, I mean. This function would be a good place to use some boolean variables, e.g.
```python
isSegmentSorted = sortedSegment in self._sortedSegments
pathContainsBranchPoints = sum(...)
if pathContainsBranchPoints and not isSegmentSorted:
    ...
```